contexts/MembershipContext/src/DataAccess/DoctrineApplicationRepository.php

```diff
@@ -236,7 +268,7 @@ private function newPaymentMethod( DoctrineApplication $application ): PaymentMe
 		}
 
 		if ( $application->getPaymentType() === PaymentType::PAYPAL ) {
-			return new PayPalPayment();
+			return new PayPalPayment( $this->newPayPalData( $application ) );
```

Comment:
Ok, using a `null` value for a constructor with default `null` will throw no type error, I've checked.